tests/cephadm/test_os_tuning_profile.py

```diff
@@ -62,16 +84,36 @@ def run(ceph_cluster, **kw):
     mount = "/tmp:/tmp"
     if config.get("command") in ("apply", "re-apply"):
         nodes = config.get("specs", {}).get("placement", {}).get("hosts")
-        hosts = [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+        if "hostx" in nodes:
+            hosts = ["NoHost1", "NoHost2"]
+        else:
+            hosts = [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
         config["specs"]["placement"]["hosts"] = hosts
         spec_file = generate_tuned_profile_spec(installler_node, config.get("specs"))
-        result = CephAdm(installler_node, mount).ceph.orch.tuned_profile.apply(
-            spec_file
-        )
-        if config.get("apply_result") != result:
-            raise OsTuningProfileError("Fail to apply tuned profile")
+        try:
+            result = CephAdm(installler_node, mount).ceph.orch.tuned_profile.apply(
+                spec_file
+            )
+            if config.get("result") not in result:
+                raise OsTuningProfileError("Fail to apply tuned profile")
+        except CommandFailed as err:
+            if config.get("result") not in str(err):
+                raise OsTuningProfileError(
+                    "apply command did not fail with the expected error"
+                )
+
+    if config.get("action") == "verify":
+        hosts = [
+            ceph_cluster.get_nodes()[int(node[-1])] for node in config.get("hosts")
+        ]
+        settings = config.get("settings")
+        result = verify_tunables(hosts, settings)
+        if result:
```

Comment:
This is not required as `verify_tunables` raising same exception in case of failure